lib/src/queue_list.dart

```diff
@@ -99,10 +99,11 @@ class QueueList<E> extends Object with ListMixin<E> implements Queue<E> {
   }
 
   QueueList<T> cast<T>() {
-    if (this is QueueList<T>) {
-      return this as QueueList<T>;
+    QueueList<Object> self = this;
+    if (self is QueueList<T>) {
+      return self;
     }
-    return QueueList.castFrom<E, T>(this);
+    return retype<T>();
   }
 
   QueueList<T> retype<T>() => QueueList.castFrom<E, T>(this);
```

Comment:
TBH, after some more consideration, I'd remove `QueueList.castFrom` again.
It's a very specific implementation class, not a general interface that others might implement, so you don't need to make the wrapping available to to others.
The reason for `Iterable.castFrom` is that it can be used to implement `Iterable.cast` for other classes that *implement* `Iterable`, but nobody should implement `QueueList`, they should just implement `Queue` and `List` if they want a class that is both.

So, consider making `castFrom` private or just inline it here.

Reply:
Made it private. Thanks!